dev/docker/README.md

```diff
@@ -0,0 +1,86 @@
+Apache REEF (incubator) Docker Tests
+====================================
+
+[Apache REEF (incubator)](http://reef.incubator.apache.org/) is 
+an approach to simplify and unify the lower layers of big data systems
+on modern resource managers. This project maintains docker images
+to simulate the target underlying systems as similar as possible.
+
+Docker-based Test Cluster
+-------------------------
+
+| OS           | Java            | Hadoop Version            | Description   | alias     |
+|--------------|-----------------|---------------------------|---------------|-----------|
+| Ubuntu 12.04 | Oracle JDK 7u80 | HDP 2.1.15 (Hadoop 2.4.0) | HDInsight 3.1 | hdi3.1    |
+| Ubuntu 12.04 | Oracle JDK 7u80 | HDP 2.2.7  (Hadoop 2.6.0) | HDInsight 3.2 | hdi3.2    |
+| Ubuntu 12.04 | Oracle JDK 7u80 | HDP 2.2.8  (Hadoop 2.6.0) |               | hdp2.2    |
+| Ubuntu 12.04 | Oracle JDK 7u80 | HDP 2.3.2  (Hadoop 2.7.1) |               | hdp2.3    |
+| Ubuntu 12.04 | Oracle JDK 7u80 | Apache Hadoop 2.7.1       |               | apache2.7 |
+| Ubuntu 12.04 | Oracle JDK 7u80 | Apache Mesos 0.24.1       |               | mesos0.24 |
+| Ubuntu 12.04 | Oracle JDK 7u80 | Apache Mesos 0.25.0       |               | mesos0.25 |
+
+Requirements
+------------
+
+First, download and *build* Apache REEF. You can find a guide for this
+[here](https://cwiki.apache.org/confluence/display/REEF/Compiling+REEF).
+Second, set $REEF_HOME to the directory of Apache REEF for the script to know its location.
+Tests will use `$REEF_HOME/lang/java/reef-tests/target/reef-tests-*-test-jar-with-dependencies.jar`.
+
+Build docker images (Optional)
+------------------------------
+
+This step is optional. We provide pre-built docker images in DockerHub.
+Please go to next step if you do not want to build images.
+To build all images manually, run `build.sh`.
+
+```sh
+$ ./build.sh
+```
+
+Run a docker-based cluster
+--------------------------
+
+Run `run-cluster.sh reefrt/{alias}`. It will create a small cluster
+with 4 nodes and run a shell on the master node.
+For all cluster, Apache Hadoop (HDFS/YARN) is installed. The hostnames
+are `hnn-001-01` (master) and `hnn-001-01 ~ hnn-001-03` (slaves).
+
+```sh
+$ ./run-cluster.sh reefrt/hdi3.2
+$ root@hnn-001-01:~#
+```
+
+Test Apache REEF on docker cluster
+----------------------------------
+
+After running a cluster, run `runyarntests.sh` for YARN-based images.
+(HDInsight, HDP, Apache YARN)
+
+```sh
+$ root@hnn-001-01:~# cd /reef
+$ root@hnn-001-01:~# ./bin/runyarntests.sh
+...
+OK (# tests)
+$ root@hnn-001-01:~# exit
+```
+
+For Mesos, run `runmesostests.sh` in a docker cluster.
```

Comment:
Could you be more specific about the image-environment mapping?
For example, `Case2: Testing Environment=Mesos images`